app/assets/javascripts/ramp_utils.js

```diff
@@ -205,10 +205,6 @@ function collapseMoreDetails() {
  */
 function disableEnableCurrentTrack(activeTrack, currentTime, isSeeked, sectionTitle) {
   let title = sectionTitle;
-  // Return when add to playlist panel is not expanded
-  if($('#addToPlaylistPanel.show').length === 0) {
-    return;
-  }
```

Comment:
Is removing this necessary for the bugfix? 

I know that there was a TypeError generated from this function when an unauthenticated user accesses the media object page, or when the "Add to Playlist" modal otherwise doesn't load. I think this prevents that error from happening? The other changes here do not seem like they would address that error being generated so this would be reintroducing it, I think.

Reply:
Yes, you are right. I had forgotten that. 

The removed code block made the current track disabling and enabling buggy when switching between a section that has tracks and without tracks. I'll see whether there's another way to achieve both these without breaking.